runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -218,6 +219,9 @@ public void resetTableData(CorfuRuntime runtime, ISerializer serializer) {
         if (streamingMapSupplier == null) {
             // PersistentCorfuTable
             oid = new ObjectsView.ObjectID(getStreamUUID(), PersistentCorfuTable.class);
+
+            // Evict all versions of this Table from MVOCache
+            runtime.getObjectsView().getMvoCache().invalidateAllVersionsOf(getStreamUUID());
```

Comment:
This will be performed even if the object is not in the ObjectView cache (i.e. even if a NoSuchElementException is thrown below). Although we don't have any such use cases today, maybe this can be moved below so that iterating through the cache is only performed if the table was in the ObjectView cache.